stdlib/public/Darwin/Accelerate/vImage_Buffer.swift

```diff
@@ -0,0 +1,348 @@
+    /// - Parameter error: Overwritten with the error code if the operation failed.
+    /// - Parameter options: The options to use when performing this operation.
+    ///
+    /// - Returns: A Core Graphics image containing a representation of the vImage buffer.
+    public func createCGImage(format: vImage_CGImageFormat,
+                              error: inout Int,
+                              flags options: vImage.Options = .noFlags) -> CGImage? {
+        var format = format
+        
+        var cgImage: CGImage!
+        
+        withUnsafePointer(to: self) {
+            cgImage = vImageCreateCGImageFromBuffer(
+                $0,
+                &format,
+                nil,
+                nil,
+                options.flags,
+                &error).takeRetainedValue()
+        }
+        
+        if error != kvImageNoError {
+            return nil
+        }
+        
+        return cgImage
+    }
+    
+    /// Creates a `CGImage` instance from a vImage buffer
+    ///
+    /// - Parameter format: The image format of this vImage buffer.
+    /// - Parameter options: The options to use when performing this operation.
+    ///
+    /// - Returns: A Core Graphics image containing a representation of the vImage buffer.
+    public func createCGImage(format: vImage_CGImageFormat,
+                              flags options: vImage.Options = .noFlags) -> CGImage? {
+        var format = format
+        var error = kvImageNoError
+        
+        var cgImage: CGImage!
+        
+        withUnsafePointer(to: self) {
+            cgImage = vImageCreateCGImageFromBuffer(
+                $0,
+                &format,
+                nil,
+                nil,
+                options.flags,
+                &error).takeRetainedValue()
+        }
+        
+        if error != kvImageNoError {
+            return nil
+        }
+        
+        return cgImage
+    }
+    
+    //===----------------------------------------------------------------------===//
+    
+    /// Copies this buffer to `destinationBuffer`.
+    ///
+    /// - Parameter destinationBuffer: The destination vImage buffer.
+    /// - Parameter options: The options to use when performing this operation.
+    ///
+    /// - Returns: `kvImageNoError`; otherwise, one of the error codes described
+    ///             in Data Types and Constants.
+    public func copy(destinationBuffer: inout vImage_Buffer,
+                     flags options: vImage.Options = .noFlags) -> vImage_Error? {
+        
+        if Int(width) == 0 {
+            return kvImageInvalidParameter
+        }
+        
+        var error = kvImageNoError
+        
+        _ = withUnsafePointer(to: self) {
+            error =  vImageCopyBuffer($0,
+                                      &destinationBuffer,
+                                      rowBytes / Int(width),
+                                      options.flags)
```

Comment:
This is unsafe in all sorts of ways that should be checked in the Swift wrapper; two off the top of my head:
- enforce that `destinationBuffer`s width and height are >= the source buffer's.
- rowBytes / width may produce zero for sub-byte pixel formats.

Reply:
I already have a check to prevent divide by zero:

        if Int(width) == 0 {
            throw vImage.Error(vImageError: kvImageInvalidParameter)
        }

I'm now throwing an error that mirrors any errors generated by vImage. So, I could add:

        if width < destinationBuffer.width || height < destinationBuffer.height {
            throw vImage.Error(vImageError: kvImageBufferSizeMismatch)
        }

...but the behavior is identical with or without it.